### DISCUSSION TITLE
Frequencies are half of what I expect

### DISCUSSION THREAD

**Discussion by sison54** [khoih-prog/TimerInterrupt#22] [Q&A] [answered]
I've tried this on both a MEGA2560 and an Uno and I get the same results. It seems that whatever frequency I set (4500Hz for example, I get around 2250Hz). I am using timer 1 and setting the frequency on the fly. Is there something I am missing? I can provide more detail if needed.

Thanks

**Comment by khoih-prog** [comment]
Thanks for your interest and bug report.

I'm able to duplicate the issue, which happens to high frequencies above around 250Hz.

**The bug has been found and fixed.** The new release will be published soon, certainly with a note for your contribution.

Regards,

**Comment by khoih-prog** [comment, marked as answer]
The [**TimerInterrupt releases v1.6.0**](https://github.com/khoih-prog/TimerInterrupt/releases/tag/v1.6.0) has just been published.

Your contribution has been noted in [**Contributions and Thanks**](https://github.com/khoih-prog/TimerInterrupt#contributions-and-thanks)

Also check and verify the new example [**Change_Interval_HF**](https://github.com/khoih-prog/TimerInterrupt/tree/master/examples/Change_Interval_HF) written for high frequencies' Timer Interrupt

---

### Releases v1.6.0

1. Fix bug resulting half frequency when using high frequencies. Check [Frequencies are half of what I expect #22](https://github.com/khoih-prog/TimerInterrupt/discussions/22)
2. Add example to demo High Frequency
3. Delete Blynk-related examples
4. Add changelog.md

**Comment by sison54** [comment]
Awesome, Thanks! Glad to know it wasn't me haha. Goes to show you if something seems wrong that it probably is lol. I sat on this for quite a while. I debated digging into to this myself to see if it was me or not, but i figured I would ask the question first because I figured you would be able to find the issue much faster.

Thanks again!

**Reply by khoih-prog** [reply]
> Glad to know it wasn't me

Be more confident next time that's you're correct, certainly after you've **investigated / tested methodologically as you did**. No one can be better than an oscilloscope in this situation.
My fault is that I thought this was designed / used / tested only for low frequencies.

Thanks to users like you :+1: , the library has been better and better, with less bugs.

**Reply by sison54** [reply]
Yup! I definitely tested as much as I could before asking, o-scope and all!. Thanks for all of your hard work!
